accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,198 @@
+
+Query to be executed: `select id from susesccrepository where url like '%SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/';`
+
+The query above will return the internal suse scc repository ID, to be used in next step.
+
+#### Insert repository authentication data
+
+- If not exists, add a new entry to table `susecredentials`, with the new authentication type `cloudrmt`, containing the repository basic authentication credentials.
+- Add a new entry to table `susesccrepositoryauth` for each repository, where:
+  - repo_id: the one obtained in previous step
+  - credentials_id: the one obtain from `susecredentials`
+  - source_id: null
+  - auth_type: `cloudrmt`
+  - auth: authentication header obtained from the PAYG instance for the repository
+
+We also need to save the repository URL, since we will not be able to use the existing SCC repository url.
+If this information will be saved in a new table field of `susesccrepositoryauth` table or in a new table is a implementation detail.
+
+
+### Implementation steps - Add product with setup wizard
+
+When a product is added in the setup wizard table `rhncontentsource` is populated according to the authentication mechanism defined in table `susesccrepositoryauth` using the repository url defined in table `susesccrepository`.
+
+We need to adapt this feature to consider the new `cloudrmt` authentication mechanism and copy the basic authentication and headers:
+  - basic authentication can be defined on repo URL (`https://user:pass@host/repo/url/`)
+  - authentication header location is an implementation detail
+    - One idea is concatenate in URL as query string parameter and adapt reposync extract and use it
+    - another possibility is using the existing mechanism, which reads from a configuration file: https://github.com/uyuni-project/uyuni/pull/2136
+
+Url to be used should be the one pointing to the cloud RMT server, saved in previous step, instead the one defined in table `susesccrepository`.
+
+### Implementation steps - reposync
+
+Reposync is already able to deal with the basic authentication mechanism (defined in repository URL).
+Http authentication headers can be read from a configuration file, specified per repository.
+Has mentioned before we need to decide how to deal with this header, if reuse the existing mechanism of implement a new one based on the database. How the header is passed from table `susesccrepositoryauth` to `rhncontentsource` and how is consume by reposync is an implementation detail.
+
+We need to maintain this table separation because reposync already have several different mechanisms relaying on this separation.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Uyuni/SUSE manager PAYG
+
+We could define a uyuni/suse manager PAYG image with access to all cloud RMT repositories.
```

Comment:
Problem: in most cases out there the Server is on-prem, managing Cloud clients.

Reply:
For it to work, I'm expecting the on-prem server needs to be on VPN to the cloud where RMT server is located, otherwise will not be able to connect it.

Reply:
Sure, there will be some kind of tunneling, I agree with that.

I am just pointing out that creating a dedicated Cloud image will not address many cases, in which the Server is on-prem.

Reply:
I have updated the text to refer to this drawback also. Thanks